runtime/onert/core/include/exec/Execution.h

```diff
@@ -69,6 +72,63 @@ class Execution
    */
   void setInput(const ir::IOIndex &index, const void *buffer, size_t length,
                 ir::Layout layout = ir::Layout::NHWC);
+
+  /**
+   * @brief     Create New IODescription instance for new inputs outputs
+   * @param[in] index   instance count number
+   */
+  void createNewAsyncDesc(uint32_t count = 0);
+
+  /**
+   * @brief     Set async input data's information
+   * @param[in] index   Input index
+   * @param[in] buffer  Input data's buffer pointer
+   * @param[in] length  Input data's length
+   * @param[in] layout  Input data's data format
+   */
+  void executeAsyncInput(const ir::IOIndex &index, const void *buffer, size_t length,
+                         ir::Layout layout = ir::Layout::NHWC);
+
+  /**
+   * @brief     Set async output data's information
+   * @param[in] index   Output index
+   * @param[in] buffer  Output data's buffer pointer
+   * @param[in] length  Output data's length
+   * @param[in] layout  Output data's data format
+   */
+  void executeAsyncOutput(const ir::IOIndex &index, void *buffer, size_t length,
+                      ir::Layout layout = ir::Layout::NHWC);
+
+  /**
+   * @brief  Async execution
+   * @note   It should be called after setting input and output buffer
+   */
+  void AsyncExecute();
+
+  /**
+   * @brief  Wait semaphores of inputs to run inference
+   */
+  void ioDescSemWait();
+
+  /**
+   * @brief  Post semaphore to manage IODescription queue order
+   */
+  void dequePost();
+
+  /**
+   * @brief  Wait semaphore to mange IODescription queue order
+   */
+  void dequeWait();
+
+  std::vector<std::tuple<std::shared_ptr<onert::exec::Execution>, onert::ir::IOIndex ,onert::ir::IOIndex>> getNextExes() { return next_exes; }
+  void pushNextExe(std::shared_ptr<onert::exec::Execution> next, onert::ir::IOIndex o_index, onert::ir::IOIndex i_index) { next_exes.push_back({next, o_index, i_index}); }
+  std::deque<std::pair<IODescription *, uint8_t>> *getAsyncIoDescs() { return &_async_io_descs; }
```

Comment:
maximum size is 255.
It would be nice to change uint32_t type instead of uint8_t